src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -118,7 +121,10 @@
           AfterTargets="CheckTestCategories"
           Inputs="@(RunTestsForProjectInputs)"
           Outputs="$(TestsSuccessfulSemaphore);$(TestPath)%(TestTargetFramework.Folder)/$(XunitResultsFileName);$(CoverageOutputFilePath)"
-          Condition="'$(TestDisabled)'!='true'">
+          >
+    <GetTargetMachineInfo>
+      <Output TaskParameter="TargetOS" PropertyName="TargetOS" />
```

Comment:
May be solveable via debugging, but when TestDisabled is true it was not getting set and producing incorrect trait stuff.  I was trying to get a PR up that contained "stuff that worked" and didn't have time to chase down all the things that get disabled when TestDisabled is true.